src/components/data/DataTable.js

```diff
@@ -318,133 +446,8 @@ const DataTable = props => {
-                          ] : [],
-                          extraColumnActions ? extraColumnActions(attributeName) : [])
-                      }, [
-                        h(HeaderCell, [
-                          !!columnNamespace && span({ style: { fontStyle: 'italic', color: colors.dark(0.75), paddingRight: '0.2rem' } }, [columnNamespace]),
-                          columnName
-                        ])
-                      ])
-                    ]),
-                    cellRenderer: ({ rowIndex }) => {
-                      const { attributes: { [attributeName]: dataInfo }, name: entityName } = entities[rowIndex]
-                      const dataCell = renderDataCell(dataInfo, workspace)
-                      const divider = div({ style: { flexGrow: 1 } })
-                      const copyButton = h(ClipboardButton, {
-                        'aria-label': `Copy attribute ${attributeName} of ${entityType} ${entityName} to clipboard`,
-                        className: 'cell-hover-only',
-                        style: { marginLeft: '1rem' },
-                        text: entityAttributeText(dataInfo)
-                      })
-                      const editLink = editable && h(EditDataLink, {
-                        'aria-label': `Edit attribute ${attributeName} of ${entityType} ${entityName}`,
-                        'aria-haspopup': 'dialog',
-                        'aria-expanded': !!updatingEntity,
-                        onClick: () => setUpdatingEntity({ entityName, attributeName, attributeValue: dataInfo })
-                      })
-
-                      if (!!dataInfo && _.isArray(dataInfo.items)) {
-                        const isPlural = dataInfo.items.length !== 1
-                        const label = dataInfo?.itemsType === 'EntityReference' ?
-                          isPlural ? 'entities' : 'entity' :
-                          isPlural ? 'items' : 'item'
-                        const itemsLink = h(Link, {
-                          style: { display: 'inline', whiteSpace: 'nowrap', marginLeft: '1rem' },
-                          onClick: () => setViewData(dataInfo)
-                        }, ` (${dataInfo.items.length} ${label})`)
-                        return h(Fragment, [dataCell, divider, copyButton, editLink, itemsLink])
-                      } else {
-                        return h(Fragment, [dataCell, divider, copyButton, editLink])
-                      }
-                    }
-                  }
-                }, visibleColumns)
-              ],
+              numFixedColumns: visibleColumns.length > 0 ? (dataProvider.features.supportsRowSelection ? 2 : 1) : 0,
+              columns: dataProvider.features.supportsRowSelection ? defaultColumns : defaultColumns.filter(column => column.field !== 'select-row-checkbox'),
```

Comment:
this works, and the end result is good. But it keeps the `dataProvider.features.supportsRowSelection ?` separate from the column it hides, and it would break if the field `select-row-checkbox` ever changed its name.

How about using the spread-operator technique (from [here](https://github.com/DataBiosphere/terra-ui/blob/8d4e385eb8106a473653a2d77ea6617dc7983f9f/src/pages/workspaces/workspace/WorkspaceContainer.js#L84)) to remove the column directly from the `defaultColumns` variable at line 312? Then this line would be just `columns: defaultColumns` (or you could move the column definitions back inline, either way works)